### DISCUSSION TITLE
Set currency externally

### DISCUSSION THREAD

**Discussion by eguvenc** [dm4t2/vue-currency-input#324] [Q&A] [answered]
I want to set currency id using a variable, but get an error , help please

Here is my code:
`                  <va-currency-input
                    v-model="agreementEditedItem.grossSalary"
                    :label="$t('employees.employeeAgreements.grossSalary')"
                    :options="{ currency: currencyId, precision: 2 }"
                  ></va-currency-input>`
        

       [Vue warn]: Error in event handler for "hook:beforeUpdate": "RangeError: Invalid currency code : null"

**Comment by dm4t2** [comment, marked as answer]
`currencyId` seems to be null, isn't it? Otherwise, the source of `<va-currency-input>` would be helpful.

**Reply by eguvenc** [reply]
No it's not null,  i set it manually.

Here is my script.
```
<template>
<div>
<v-row>
<va-currency-input
    v-model="agreementEditedItem.grossSalary"
    :options="{ currency: currencyId, precision: 2 }"
  ></va-currency-input>
</div>
</row>
</template>

export default {
  data() {
    return {
      currencyId: 'USD',
      agreementEditedItem: {
          grossSalary: null
       }
    }
   }
}
```

And below the codes source of va-currency-input
```
<template>
  <v-text-field
    v-bind="commonProps"
    :filled="filled"
    ref="inputRef" 
    @change="changeState"
  ></v-text-field>
</template>

<script>
import { useCurrencyInput } from 'vue-currency-input'
import Input from "../../../mixins/input";
import Editable from "../../../mixins/editable";

export default {
  mixins: [Input, Editable],
  props: {
    /**
     * Number to be edited.
     * @model
     */
    value: {
      // type: Number,
      default: null,
    },
    /**
     * Use full filled background color style.
     */
    filled: {
      type: Boolean,
      default: true,
    },
    options: Object
  },
  methods: {
    changeState(value) {
        this.change(value)
        this.update(value, true)
    },
  },
  setup(props) {
    const { inputRef } = useCurrencyInput(props.options)
    return { inputRef }
  }
};
</script>
```

**Reply by dm4t2** [reply]
I can only guess, but you mixed up both Composition API and Options API (mixins) in your component, probably causing your issue.
A working example for Vue2+Vuetify can be found in the [Vue2 example repo](https://github.com/dm4t2/vue-currency-input/blob/master/examples/vue2/src/components/VCurrencyField.vue).

**Reply by eguvenc** [reply]
It works fine!  Thanks.....
